### PR TITLE
[ISSUE #3313] Migrated public chat creation flow

### LINKED ISSUE #3313: Join existing public chat or create a new channel
### User Story

As a user, I want to view a featured list of public chats so that I can join them and start chatting.

As a user, I want to create a public chat so that I can create a community and start chatting.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

*Summary*: From Home hit `New`, then `Join public chat` and either a) enter a channel name to create a topic, or tap on a topic to join an existing public chat.

[Zeplin designs](https://app.zeplin.io/project/5a55d587ba304eed4d2c942f/screen/5a70317c76365a893ce40c22).

*Topics*:
* #status 
* #openbounty  
* #ethereum

### LINKED ISSUE #3341: Destructive actions (delete, leave chat) should ask for confirmation before execution [3308]
### Description

*Type*: Bug

*Summary*: Destructive actions (delete chat, leave chat) should ask for confirmation before execution. Currently, if user taps on Delete or Leave chat in the chat menu then action is triggered immediately and user has no possibility to "undo" it.

#### Expected behavior
If any destructive action is triggered then confirmation should be shown first to check is it intentional
Design for confirmation:
36161924-0036293c-10f6-11e8-8553-dcb96492a2b2

#### Actual behavior
Delete/Leave group chat is executed immediately

### Reproduction

- Open Status
- Open Group or public chat
- Open menu (3 dots in top right corner)
- Tap on Delete chat or on Leave Group chat

### Additional Information
* Status version: PR https://github.com/status-im/status-react/pull/3308
* Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by jeluard**
fixes #3313 
fixes #3341 

Note that input errors have not been migrated to new paradigm.

### Summary:

Migrated public chat creation flow to latest mockups. Also added popup prompt to destructive actions (applies to group chat too)

### Steps to test:
- Open Status
- Navigate home tab
- Create a public chat: validates ii follows the [mockups](https://app.zeplin.io/project/5a55d587ba304eed4d2c942f/screen/5a70317c76365a893ce40c22)

status: ready

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

File: src/status_im/chat/events.cljs [hunk 1]
```diff
@@ -347,3 +343,19 @@
...
+
+(handlers/register-handler-fx
+  :delete-chat
+  [re-frame/trim-v]
+  (fn [cofx [chat-id]]
+    (-> (models/remove-chat cofx chat-id)
+        (update :db navigation/replace-view :home))))
+
+(handlers/register-handler-fx
+  :delete-chat?
```
Comment (flexsurfer) [inline src/status_im/chat/events.cljs:353]: maybe `:show-delete-chat-confirmation`

File: src/status_im/chat/events.cljs [hunk 2]
```diff
@@ -347,3 +343,19 @@
...
+  [re-frame/trim-v]
+  (fn [_ [chat-id group?]]
+    {:show-confirmation {:title               (str (i18n/label :t/delete) "?")
```
Comment (flexsurfer) [inline src/status_im/chat/events.cljs]: is "?" using in all languages?

File: src/status_im/chat/handlers.cljs [hunk 3]
```diff
@@ -32,10 +27,18 @@
                       :private private-key}
            :message  {:from       current-public-key
                       :message-id (random/id)}})))
-     (dispatch [:remove-chat current-chat-id]))))
+     (re-frame/dispatch [:remove-chat current-chat-id]))))
+
+(handlers/register-handler-fx
+  :leave-group-chat?
```
Comment (flexsurfer) [inline src/status_im/chat/handlers.cljs]: maybe `:show-leave-group-chat-confirmation` ?

File: src/status_im/ui/screens/group/chat_settings/events.cljs [hunk 4]
```diff
@@ -162,14 +161,22 @@
...
 
-(register-handler-fx
+(handlers/register-handler-fx
   :clear-history
   (fn [{{:keys [current-chat-id] :as db} :db} _]
     {:db (assoc-in db [:chats current-chat-id :messages] {})
      :delete-messages current-chat-id}))
+
+(handlers/register-handler-fx
+  :clear-history?
```
Comment (flexsurfer) [inline src/status_im/ui/screens/group/chat_settings/events.cljs:177]: `show-clear-history-confirmation` ?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/chat/events.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/chat/events.cljs:353]: We don't use `?` yet but could be a convention for handler name? `-confirmation` looks pretty long :)

File: src/status_im/chat/events.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/chat/events.cljs]: @jeluard ?

Comment (jeluard) [reply src/status_im/chat/events.cljs]: Oh my bad, good point. Will make this label too.

**Comment by asemiankevich** [comment]
Branch: PR-3381
Android: https://i.diawi.com/XsniRz
iOS: https://i.diawi.com/jMnfm8

**Comment by asemiankevich** [comment]
1. input field on iOS does not show what I am typing into it. 

image

2. joining status public chat throws error 

image

3. joining other pre-defined public chats showing Unknown (that may be due existing bug in develop) and dApp chat is shown on Home tab after I join public chat

image


image

**Comment by asemiankevich** [comment]
Please also add accessibility lables here cc:  @lukaszfryc

**Comment by lukaszfryc** [comment]
To follow up on @asemiankevich comment -- we want to kick off iOS end-end automation and to do it efficiently we need accessibility labels for elements user may interact with (buttons, list items, menu options, etc). Whenever UI changes we should consider updating the labels. 

Here is example of accessibility labels we want to introduce in another part of the app #3394

**Comment by asemiankevich** [comment]
all above issues are fixed, 1 more minor issue - wrong alignment for input in iOS

image

**Comment by jeluard** [comment]
@asemiankevich Great thanks! Can we left the alignment issue out? I will fix it part of #3372

**Comment by asemiankevich** [comment]
@jeluard yes, for sure!

**Pull request merged**